kart/helper.py

```diff
@@ -169,6 +170,16 @@ def helper(ctx, socket_filename, timeout, args):
                         f"Payload:\n{repr(payload)}",
                     )
                 else:
+                    try:
+                        # Join the process group of the calling process - so that if they get killed, we get killed to.
+                        os.setpgid(0, calling_environment["pid"])
+                        os.environ["_KART_PGID_SET"] = "1"
+                    except Exception as e:
```

Comment:
I suspect this could catch a more specific exception class (OSError? or one of the subclasses: https://docs.python.org/3/library/exceptions.html#os-exceptions)

Reply:
Yes, the [possible error codes from `setpgid()`](https://man7.org/linux/man-pages/man2/getpgid.2.html#ERRORS) are all mapped to subclasses of `OSError`.

Reply:
Done